### PR TITLE
added subscribe notification characteristic example

### PR DESCRIPTION
**Pull request by blandger**
I worked on that missing and main example for a long time since sync api.
Rebease didn't succeed. New fork and new PR.

### CODE REVIEW

**Comment by blandger** [comment]
That seems to me current build is bull shit, imho.

**Comment by blandger** [comment]
I added back 'print adapter' because it's not possible to print adapter info on Windows.

File: examples/discover_adapters_peripherals.rs [hunk 1]
```diff
@@ -30,37 +30,44 @@ async fn main() -> Result<(), Box<dyn Error>> {
...
-                            properties.local_name
+                            &local_name
                         );
-                        peripheral
+                        let connect_result = peripheral
                             .connect()
-                            .await
-                            .expect("Can't connect to peripheral...");
+                            .await;
+                        match connect_result {
```
Comment (qwandor) [inline examples/discover_adapters_peripherals.rs:47]: Just use `if let` for this.

File: examples/subscribe_notify_characteristic.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    let rdr = Cursor::new(data.value);
+    println!("Received data from [{:?}] = {:?}", data.uuid, rdr);
+}
+
+/**
+If you are getting run time error like that :
+ thread 'main' panicked at 'Can't scan BLE adapter for connected devices...: PermissionDenied',
+ you can try to run app with > sudo ./discover_adapters_peripherals
+ on linux
+**/
```
Comment (qwandor) [inline examples/subscribe_notify_characteristic.rs:35]: This comment is no longer relevant for the current Linux implementation, remove it.

File: examples/subscribe_notify_characteristic.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    println!("peripheral : {:?} is connected: {:?}",
+                             &local_name, is_connected);
+                    // filter needed peripheral
+                    if local_name.contains(PERIPHERAL_NAME_MATCH_FILTER) {
+                        println!("found matching peripheral : {:?}...", &local_name);
+                        if !is_connected { // we can be connected previously
+                            let connect_result = peripheral
+                                .connect()
+                                .await;
+                            match connect_result {
```
Comment (qwandor) [inline examples/subscribe_notify_characteristic.rs:72]: Again, just `if let` is simpler.

File: examples/subscribe_notify_characteristic.rs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                        }
+                        let is_connected = peripheral.is_connected().await?;
+                        println!(
+                            "now connected (\'{:?}\') to peripheral : {:?}...",
+                            is_connected, &local_name
+                        );
+                        let chars = peripheral.discover_characteristics().await?;
+                        if is_connected {
+                            println!(
+                                "Discover peripheral : \'{:?}\' characteristics...",
```
Comment (qwandor) [inline examples/subscribe_notify_characteristic.rs:88]: Won't this result in the name being quoted twice? Get rid of the extra quote marks.

**Review by qwandor — APPROVED** [review]

**Pull request merged**